lib/transport/selenium-webdriver/browserstack/automate.js

```diff
@@ -0,0 +1,25 @@
+const BrowserStack = require('./browserstack.js');
+const AppiumMixin = require('../appiumMixin.js');
+const {Capabilities} = require('selenium-webdriver');
+
+class Automate extends BrowserStack {
+  get ApiUrl() {
+    return `https://api.browserstack.com/${this.productNamespace}`;
+  }
+
+  get productNamespace() {
+    return 'automate';
+  }
+
+  createDriver({options = this.desiredCapabilities}) {
+    if (options instanceof Capabilities) {
+      return super.createDriver({options});
+    }
+
+    return this.createAppiumDriver({options});
+  }
+}
+
+Object.assign(Automate.prototype, AppiumMixin);
```

Comment:
can't we use normal inheritance here?

Reply:
We could but then `browserstack.js` would have to extend to `appium.js` because we need a few functionalities from the `AppiumServer` class in automate and app-automate and then to make it work, we'd have to write some additional code in `appium.js` related to BrowserStack. Commit [e000786](https://github.com/nightwatchjs/nightwatch/pull/3519/commits/e000786fc403ddaf6f9bd79b25ce0549da4e8880) does just that.

But then, as I think now, `browserstack.js` and `appium.js` should be more or less independent and we shouldn't need to write additional code in `appium.js` to support `browserstack.js`, because otherwise when `appium.js` would grow at some later point of time, we'd always have to do something extra to ensure that browserstack still works fine.

Reply:
or both browserstack and appium could extend a base class.

Reply:
Please check [0706064](https://github.com/nightwatchjs/nightwatch/pull/3519/commits/0706064a1a590ab9ae47d787d22a5e189ebd39d5).